_context.py: make directories in the configuration absolute

abderrahim opened the pull request:
bst doesn't check that the paths set are absolute and sometimes breaks if
they aren't

bst 1 backport of #1440

gtristan commented:
Will this approach help us run multiple parallel containers for the sake of the remote execution tests ?

Overall it looks good, but I think we should update the `merge.yml` and `release.yml` workflows to build the docs this way also.

cs-shadow commented:
> Will this approach help us run multiple parallel containers for the sake of the remote execution tests ?

Indeed. Compose supports bringing up multiple services, have dependencies across them etc so it should be reasonably straightforward to add remote executetion tests. In fact, the old remote execution tests already were using such a compose file: https://gitlab.com/BuildStream/buildstream/-/blob/master/.gitlab-ci/buildgrid-remote-execution.yml

> Overall it looks good, but I think we should update the merge.yml and release.yml workflows to build the docs this way also.

Thanks! I'll fix this and fix a couple of other reamining TODOs later today.

cs-shadow commented:
This should now be all rebased and ready to go. The two remaining workflows have now been updated.

abderrahim committed:
_context.py: make directories in the configuration absolute

bst doesn't check that the paths set are absolute and sometimes breaks if
they aren't
